tests/serverpod_test_server/test/generated_methods/from_json_test.dart

```diff
@@ -0,0 +1,289 @@
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test(
+      'When deserializing a class and providing null value for a non-nullable value in JSON, it should throw a TypeError.',
+      () {
+    expect(
+      () => SimpleData.fromJson({"num": null}),
+      throwsA(isA<TypeError>()),
+    );
+  });
+
+  test(
+      'When deserializing a class which has non-nullable types and providing a empty JSON, it should throw a TypeError.',
+      () {
+    expect(
+      () => SimpleData.fromJson({}),
+      throwsA(isA<TypeError>()),
+    );
+  });
+
+  test(
+      'When deserializing a class with non-nullable values and providing correct values in JSON, it should not throw a TypeError.',
+      () {
+    expect(
+      SimpleData.fromJson({
+        "id": 1,
+        "num": 1,
+      }),
+      isA<SimpleData>(),
+    );
+  });
+
+  test(
+      'When deserializing a class with non-nullable types but providing different value types in JSON, it should throw a TypeError.',
+      () {
+    expect(
+      () => SimpleData.fromJson({
+        "id": "test",
+        "num": "test",
+      }),
+      throwsA(isA<TypeError>()),
+    );
+  });
+
+  test(
+      'When deserializing a class with non-nullable DateTime and correct value provided in JSON, expecting values to be deserialized correctly.',
+      () {
+    var types = Types.fromJson({
+      'aDateTime': '2024-01-01T00:00:00.000Z',
```

Comment:
Suggestion: Missing test for an incorrect datetime deserialization.

Also missing test for when a parsed DateTime is passed in. According to our code these should be accepted by the deserialization.

Reply:
for this one i also added 'extensions_test.dart' on the serverpod_serialization pacakge.